crates/red_knot_python_semantic/src/types.rs

```diff
@@ -233,31 +233,39 @@ fn declarations_ty<'db>(
 /// Unique ID for a type.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
 pub enum Type<'db> {
-    /// the dynamic type: a statically-unknown set of values
+    /// The dynamic type: a statically-unknown set of values
     Any,
-    /// the empty set of values
+    /// The empty set of values
     Never,
-    /// unknown type (either no annotation, or some kind of type error)
-    /// equivalent to Any, or possibly to object in strict mode
+    /// Unknown type (either no annotation, or some kind of type error).
+    /// Equivalent to Any, or possibly to object in strict mode
     Unknown,
-    /// name does not exist or is not bound to any value (this represents an error, but with some
+    /// Name does not exist or is not bound to any value (this represents an error, but with some
     /// leniency options it could be silently resolved to Unknown in some cases)
     Unbound,
-    /// the None object -- TODO remove this in favor of Instance(types.NoneType)
+    /// The None object -- TODO remove this in favor of Instance(types.NoneType)
     None,
-    /// a specific function object
+    /// Temporary type for symbols that can't be inferred yet because of missing implementations.
+    /// This variant should eventually be removed once red-knot is spec-compliant.
+    ///
+    /// General rule: `Todo` should only propagate when the presence of the input `Todo` caused the
+    /// output to be unknown. An output should only be `Todo` if fixing the input to be correctly
+    /// inferred would cause the output to be known - this is not the case with `[Type::Any]` or
+    /// `[Type::Unknown]`.
```

Comment:
I'd use a simpler wording here. The exact interaction with Any or Unknown depends on the specific case.

```suggestion
    /// output to be unknown. An output should only be `Todo` if fixing all `Todo` inputs to be not `Todo` would change the output type.
```